script/Deploy.s.sol

```diff
@@ -7,15 +7,16 @@ import { StakeManager } from "../contracts/StakeManager.sol";
 import { VaultFactory } from "../contracts/VaultFactory.sol";
 
 contract Deploy is BaseScript {
-    function run() public returns (VaultFactory, StakeManager, DeploymentConfig) {
+    function run() public returns (VaultFactory, StakeManager, StakeManager, DeploymentConfig) {
         DeploymentConfig deploymentConfig = new DeploymentConfig(broadcaster);
         (, address token) = deploymentConfig.activeNetworkConfig();
 
         vm.startBroadcast(broadcaster);
         StakeManager stakeManager = new StakeManager(token, address(0));
+        StakeManager migrationStakeManager = new StakeManager(token, address(stakeManager));
```

Comment:
This needs clarification.

This looks like you're adding a deployment of a second stakemanager to the deployment script, just so you have a second stakemanager instance in your tests.

If this is the case, then we should remove this change here.
Remember that the deploy scripts are used for deployment (they also just happen to be used within tests, because we're then also testing our deployment scripts implicitly).

We probably don't want to deploy two stakemanagers every single time.
In fact, we probably want a **new** deployscript that simply puts out a new stakemanager with a reference to an older one, but without deploying the original one, nor the vaultfactory.

Actually, I'll create an issue for this.

If you need an additional stakemanager in your tests, I'd say, either deploy it there directly, or add a new deploy script that does what I've described above.

^ This is all based on the assumption I've laid out in the beginning. If this doesn't apply, please let me know.